uisdk/src/main/java/tech/dojo/pay/uisdk/domain/mapper/PaymentIntentDomainEntityMapper.kt

```diff
@@ -3,22 +3,57 @@ package tech.dojo.pay.uisdk.domain.mapper
 import tech.dojo.pay.sdk.card.presentation.gpay.util.centsToString
 import tech.dojo.pay.uisdk.data.entities.PaymentIntentPayload
 import tech.dojo.pay.uisdk.domain.entities.AmountDomainEntity
-import tech.dojo.pay.uisdk.domain.entities.EssentialParamMissingException
 import tech.dojo.pay.uisdk.domain.entities.ItemLinesDomainEntity
 import tech.dojo.pay.uisdk.domain.entities.PaymentIntentDomainEntity
 import tech.dojo.pay.uisdk.domain.entities.PaymentIntentStatusDomainEntity
+import java.util.Currency
 
 internal class PaymentIntentDomainEntityMapper {
-    fun apply(raw: PaymentIntentPayload): PaymentIntentDomainEntity {
-        checkInvalidParameters(raw)
-        return PaymentIntentDomainEntity(
+    fun apply(raw: PaymentIntentPayload): PaymentIntentDomainEntity? {
```

Comment:
Do you mind renaming this function to `mapPayload` or something else? everytime I read `.apply` I think it is the kotlin function and this is confusing 😅